src/System.Memory/src/System/Buffers/ReadOnlyBufferExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Extension methods for <see cref="ReadOnlyBuffer{T}"/>
+    /// </summary>
+    public static class ReadOnlyBufferExtensions
+    {
+        /// <summary>
+        /// Returns position of first occurance of item in the <see cref="ReadOnlyBuffer{T}"/>
+        /// </summary>
+        public static SequencePosition? PositionOf<T>(this ReadOnlyBuffer<T> buffer, T value) where T : IEquatable<T>
+        {
+            SequencePosition position = buffer.Start;
+            SequencePosition result = position;
+            while (buffer.TryGet(ref position, out var memory))
```

Comment:
I was confused by this loop since we keep setting result to position and neither seem to change. Just realized this method has a default `advance = true` argument. :)